api/app/signals/apps/signals/models/area.py

```diff
@@ -0,0 +1,45 @@
+from django.db.models import F, Max, Min
+from django.contrib.gis.db import models
+
+# TODO: slugfields for the "code" fields below
+
+
+class AreaManager(models.Manager):
+    # Only show the most recent properties for an area
+    def get_queryset(self):
+        qs = super().get_queryset().annotate(
+            updated_at=Max('properties__created_at'),
+            created_at=Min('properties__created_at'),
+        ).filter(
+            properties__created_at=F('updated_at')
+        )
+        return qs
+
+
+class AreaType(models.Model):
+    name = models.CharField(max_length=255)
+    code = models.CharField(max_length=255)
+    description = models.TextField(max_length=3000)
+
+
+# Instances of Area are not to be edited after creation, in stead, their
+# editable properties should be set by appendinng an AreaProperties instance
+# pointing towards the relevant Area instance.
+class Area(models.Model):
+    class Meta:
+        unique_together = ['code', 'type']
+
+    code = models.CharField(max_length=255)
+    type = models.ForeignKey(AreaType, on_delete=models.CASCADE)
```

Comment:
Do we want to keep track of the history of the type of an area? And if so do we want to do that the same way as we do for the properties?

Reply:
The idea was that you can make a foreign key to an `Area` instance, and that that relation will remain constant (even though some of the properties of an Area are changed).

We could consider adding `type` to the tracked properties.